strawberry/experimental/pydantic/object_type.py

```diff
@@ -91,34 +92,36 @@ def wrap(cls):
             if name in fields_set
         ]
 
-        cls_annotations = getattr(cls, "__annotations__", {})
+        wrapped = _wrap_dataclass(cls)
+        extra_fields = _get_fields(wrapped)
+
         all_fields.extend(
             (
                 (
-                    name,
-                    type_,
-                    StrawberryField(
-                        python_name=name,
-                        graphql_name=None,
-                        type_annotation=type_,
-                        # we need a default value when adding additional fields
-                        # on top of a type generated from Pydantic, this is because
-                        # Pydantic Optional fields always have None as default value
-                        # which breaks dataclasses generation; as we can't define
-                        # a field without a default value after one with a default value
-                        # adding fields at the beginning won't work as we will also
-                        # support default values on them (so the problem will be just
-                        # shifted around)
-                        default=None,
-                    ),
+                    field.name,
+                    field.type,
+                    field,
                 )
-                for name, type_ in cls_annotations.items()
+                for field in extra_fields
             )
         )
 
+        # Sort fields so that fields with missing defaults go first
+        # because dataclasses require that fields with no defaults are defined
+        # first
```

Comment:
should we add a test for this case?

Reply:
The changes I made to `test_basic` already cover this I think (the order changed)

Reply:
oh man, I just realised this is just for Pydantic :D